src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -0,0 +1,273 @@
+                            sendCompleteEventToKafka(tenantId, sharingInstance, kafkaConfig, record.headers());
+                            promise.complete();
+                          }).onFailure(error -> {
+                            String errorMessage = format("Error update Instance by id %s on the source tenant %s. Error: %s", instanceId, sharingInstance.getTargetTenantId(), error.getCause());
+                            sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+                            promise.fail(error);
+                          });
+                      })
+                      .onFailure(publishFailure -> {
+                        String errorMessage = format("Error save Instance by id %s on the target tenant %s. Error: %s", instanceId, sharingInstance.getTargetTenantId(), publishFailure.getCause());
+                        sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+                        promise.fail(publishFailure);
+                      });
+                  } else if ("MARC".equals(srcInstance.getSource())) {
+                    String errorMessage = format("Error sharing Instance with id %s to the target tenant %s. Because source is %s",
+                      instanceId, sharingInstance.getTargetTenantId(), srcInstance.getSource());
+                    sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+                    promise.fail(errorMessage);
+                  } else {
+                    String errorMessage = format("Error sharing Instance with id %s to the target tenant %s. Because source is %s",
+                      instanceId, sharingInstance.getTargetTenantId(), srcInstance.getSource());
+                    sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+                    promise.fail(errorMessage);
+                  }
+                }
+              })
+              .onFailure(err -> {
+                String errorMessage = format("Error retrieving Instance by id %s from source tenant %s. Error: %s", instanceId, sharingInstance.getSourceTenantId(), err);
+                sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+                promise.fail(errorMessage);
+              });
+          } else {
+            String errorMessage = format("Error checking Instance by id %s on target tenant %s. Error: %s", instanceId, sharingInstance.getTargetTenantId(), failure.getMessage());
+            sendErrorResponseAndPrintLogMessage(tenantId, errorMessage, sharingInstance, record.headers());
+            promise.fail(errorMessage);
+          }
+        })
+        .onSuccess(instanceOnTargetTenant -> {
+          String errorMessage = format("Instance %s is present on target tenant: %s", instanceId, sharingInstance.getTargetTenantId());
```

Comment:
if the instance is already present, I think we should just send a complete message without error.

Reply:
I sent a completed message with a warning description.